src/Rings/mpoly-graded.jl

```diff
@@ -2184,66 +2188,66 @@ end
-    end
-    if is_empty(gens(I))  # special handling for ideal with no gens (?is there a cleaner way?)
-        tmp = homogenization(one(P), W, var; pos=pos)
-        R = parent(tmp)
-        return ideal(R) # no gens, so it is the zero ideal
-    end # of special handling for zero ideal
-    Hgens = homogenization(_gens_for_homog_via_sat(I, extra_gens_flag), W, var; pos=pos)
-    R = parent(Hgens[1])
-    DoSatByProduct = false  # true means saturate by product of homogenizing vars;
-                            #false means saturate successively by each homogenizing var (probably faster, in most cases?)
-    if DoSatByProduct
-        prod_h_vars = prod([gen(R,i)  for i in pos:(pos+size(W, 1)-1)])  # product of the homogenizing variables
-        Ih = saturation(ideal(R, Hgens), ideal(R, prod_h_vars))
-    else # not DoSatByProduct, so do a cascade of saturations
-        Ih = ideal(R, Hgens)
-        for i in pos:(pos+size(W, 1)-1)
-            Ih = saturation(Ih, ideal(R, gen(R,i)))
-        end
+  P = base_ring(I)
+  if pos === nothing
+    pos = 1+ngens(P)
+  else
+    @req  pos in 1:1+ngens(P)  "Homog var index out of range."
+  end
+  if is_empty(gens(I))  # special handling for ideal with no gens (?is there a cleaner way?)
+    tmp = homogenization(one(P), W, var; pos=pos)
+    R = parent(tmp)
+    return ideal(R) # no gens, so it is the zero ideal
+  end # of special handling for zero ideal
+  Hgens = homogenization(_gens_for_homog_via_sat(I, extra_gens_flag), W, var; pos=pos)
+  R = parent(Hgens[1])
+  DoSatByProduct = false  # true means saturate by product of homogenizing vars;
+  #false means saturate successively by each homogenizing var (probably faster, in most cases?)
+  if DoSatByProduct
+    prod_h_vars = prod([gen(R,i)  for i in pos:(pos+size(W, 1)-1)])  # product of the homogenizing variables
+    Ih = saturation(ideal(R, Hgens), ideal(R, prod_h_vars))
+  else # not DoSatByProduct, so do a cascade of saturations
+    Ih = ideal(R, Hgens)
+    for i in pos:(pos+size(W, 1)-1)
+      Ih = saturation(Ih, ideal(R, gen(R,i)))
     end
-    return Ih
-######  2023-07-14: (this comment is now old, but probably still valid)
-######  There is a problem/bug with Singular.satstd: so disable this version
-######  IH = ideal(R, Hgens);  Y = ideal(R, prod_h_vars);
-######  singular_assure(IH); singular_assure(Y); return Singular.satstd(IH.gens.S, Y.gens.S)
+  end
+  return Ih
+  ######  2023-07-14: (this comment is now old, but probably still valid)
+  ######  There is a problem/bug with Singular.satstd: so disable this version
+  ######  IH = ideal(R, Hgens);  Y = ideal(R, prod_h_vars);
+  ######  singular_assure(IH); singular_assure(Y); return Singular.satstd(IH.gens.S, Y.gens.S)
 end
 
 
 # ============================================
 # 2023-06-30 START: New impl of homogenization
-# By John Abbott, based on K+R Book vol 2
+# By John Abbott, based on K+R Book "Computational Commutative Algebra" (see "justification" below)
```

Comment:
This PR seems to contain other changes beyond those advertised in the title?

It also has conflicts now...

Did you generate this with a script? If so, perhaps we can just redo this cleanly from latest `master`, with the script in the commit message (handy for future repetitions of this change)

Reply:
Should I look to see how scary "resolve conflicts" is?

Reply:
It seems to be mostly due to https://github.com/oscar-system/Oscar.jl/pull/2630 and https://github.com/oscar-system/Oscar.jl/pull/2614.